File: Algorithms/Implementation/Kangaroo.java
package Implementation;

import java.util.Scanner;

/**
 *	Algorithms -> Warmup -> Kangaroo
 *	Easy (Super Easy)
 */
public class Kangaroo {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		double d0 = in.nextInt();
    	double v0 = in.nextInt();
    	double d1 = in.nextInt();
    	double v1 = in.nextInt();
        in.close();
        String r = willKagarooCatchUp(d0, v0, d1, v1);
        System.out.println(r);
	}
	
	public static String willKagarooCatchUp(double d0, double v0, double d1, double v1)
	{
		double i = (d0 - d1) / (v1 - v0);
    	String r = "NO";
    	if(i > 0 && i == (int) i) {
    		r = "YES";
    	}
    	return r;
	}
}

File: Algorithms/Implementation/Kangaroo.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

/**
 *	Algorithms -> Warmup -> Kangaroo
 *	Easy
 */

struct Kangaroo {
	int position;
	int velocity;
} kangaroo;

/*
 * Function declaration
 */
int willSecondKangarooCatchTheFirst(struct Kangaroo k1, struct Kangaroo k2);

/*
 * Main
 */
int main(){

	struct Kangaroo k1;
	struct Kangaroo k2;

	scanf("%d %d %d %d", &k1.position, &k1.velocity, &k2.position, &k2.velocity);

	int result = willSecondKangarooCatchTheFirst(k1, k2);

	if(result == 1) {
		printf("YES");
	} else {
		printf("NO");
	}

    return 0;
}

/*
 * Functions
 */
int willSecondKangarooCatchTheFirst(struct Kangaroo k1, struct Kangaroo k2) {

	double i = (double) (k1.position - k2.position) / (k2.velocity - k1.velocity);

	int result = 0;
	if(i > 0 && i == (int) i) {
		result = 1;
	}
	return result;
}
